README.md

<!-- README.md is generated from README.Rmd. Please edit that file -->

### **partiesAL**: Legislative Parties of the Americas 1925 - 2019

*Aníbal Pérez-Liñán, Nicolás Schmidt, Daniela Vairo*

<!-- badges: start -->

[![](https://img.shields.io/badge/devel%20version-0.1.0-orange.svg)](https://github.com/Nicolas-Schmidt/partiesAL)
[![Project Status: Active – The project has reached a stable, usable
state and is being actively
developed.](https://www.repostatus.org/badges/latest/active.svg)](https://www.repostatus.org/#active)

<!-- badges: end -->

## Description

This package provides two related datasets. The first one, called
**`parties`**, is a dataset where the unit of analysis is the political
party in each year in the congress for 21 countries of the Americas (20
Latin American countries plus the United States) in the period from 1925
to 2019. The second dataset (**`parties2`**) is an aggregation of this
data, compiled into a unit that becomes country-year. This second
database provides an additional set of systemic variables such as the
effective number of parties in each chamber of the congress.

The countries included in the database are:

``` r
unique(partiesAL::parties$ccode)
#>  [1] "ARG" "BOL" "BRA" "CHL" "COL" "CRI" "CUB" "DOM" "ECU" "GTM" "HND" "HTI"
#> [13] "MEX" "NIC" "PAN" "PER" "PRY" "SLV" "URY" "VEN" "USA"
```

- There are two ways to access the data. One option is to install this R
  package (`partiesAL`) by following the detailed instructions below for
  a conventional installation from a repository. The second option, for
  users who prefer to access the data outside of R, provides three
  available choices:
  [**csv**](https://github.com/Nicolas-Schmidt/partiesAL/tree/master/man/figures/export/csv),
  [**xlsx**](https://github.com/Nicolas-Schmidt/partiesAL/tree/master/man/figures/export/xlsx),
  [**dta**](https://github.com/Nicolas-Schmidt/partiesAL/tree/master/man/figures/export/dta).

- The codebook for the databases can be accessed through this
  [**link**](https://github.com/Nicolas-Schmidt/partiesAL/tree/master/man/figures).

## `parties` and `parties2`

The structure of the databases is as follows

``` r
str(partiesAL::parties)
#> tibble [11,822 × 22] (S3: tbl_df/tbl/data.frame)
#>  $ cowcode  : num [1:11822] 160 160 160 160 160 160 160 160 160 160 ...
#>  $ ccode    : chr [1:11822] "ARG" "ARG" "ARG" "ARG" ...
#>  $ year     : num [1:11822] 1925 1925 1925 1925 1925 ...
#>  $ legis    : num [1:11822] 1 1 1 1 1 1 1 1 1 1 ...
#>  $ date_low : POSIXct[1:11822], format: "1924-03-02" "1924-03-02" ...
#>  $ date_upp : POSIXct[1:11822], format: "1925-03-01" "1925-03-01" ...
#>  $ pty_acrn : chr [1:11822] "UCR" "CONS" "UCR" "PS" ...
#>  $ pty_name : chr [1:11822] "Unión Cívica Radical" "Conservative" "Unión Cívica Radical" "Socialista" ...
#>  $ pty_code : num [1:11822] 1.60e+08 1.60e+08 1.60e+08 1.60e+08 1.61e+08 ...
#>  $ fac_name : chr [1:11822] "Radical Antipersonalista" NA "Radical Yrigoyenist" "Socialist" ...
#>  $ fac_code : chr [1:11822] "160189001001" NA "160189001002" "160189601002" ...
#>  $ s_low    : num [1:11822] 3 14 72 18 17 14 15 31 40 55 ...
#>  $ ts_low   : num [1:11822] 158 158 158 158 158 158 158 158 158 158 ...
#>  $ s_upp    : num [1:11822] 13 8 4 2 0 0 0 13 8 4 ...
#>  $ ts_upp   : num [1:11822] 30 30 30 30 30 30 30 30 30 30 ...
#>  $ presp    : num [1:11822] 1 0 0 0 0 0 0 1 0 0 ...
#>  $ cl_other : num [1:11822] 1 0 0 0 1 0 0 1 0 0 ...
#>  $ cl_altman: num [1:11822] NA NA NA NA NA NA NA NA NA NA ...
#>  $ cl_deheza: num [1:11822] NA NA NA NA NA NA NA NA NA NA ...
#>  $ cl_dpi   : num [1:11822] NA NA NA NA NA NA NA NA NA NA ...
#>  $ founded  : num [1:11822] 1890 1874 1890 1896 8888 ...
#>  $ source   : chr [1:11822] "PHW 1928, p.3/(Gibson 1996, 40)" "PHW 1928, p.3/(Gibson 1996, 40)" "PHW 1928, p.3/(Gibson 1996, 40)" "PHW 1928, p.3/(Gibson 1996, 40)" ...
```

``` r
str(partiesAL::parties2)
#> 'data.frame':    1995 obs. of  17 variables:
#>  $ cowcode  : num  160 160 160 160 160 160 160 160 160 160 ...
#>  $ ccode    : chr  "ARG" "ARG" "ARG" "ARG" ...
#>  $ year     : num  1925 1926 1927 1928 1929 ...
#>  $ legis    : num  1 1 1 1 1 1 0 1 1 1 ...
#>  $ newleg   : num  1 1 0 1 0 1 0 1 0 1 ...
#>  $ date_h   : POSIXct, format: "1924-03-02" "1926-03-07" ...
#>  $ year_h   : num  1924 1926 1926 1928 1928 ...
#>  $ month_h  : num  3 3 3 4 4 4 NA 11 11 3 ...
#>  $ date_s   : POSIXct, format: "1925-03-01" "1925-03-01" ...
#>  $ p_h      : num  0.0196 0.2095 0.2095 0.5613 0.5613 ...
#>  $ p_s      : num  0.481 0.481 0.481 0.269 0.269 ...
#>  $ g_h      : num  0.131 0.209 0.209 0.561 0.561 ...
#>  $ g_s      : num  0.481 0.481 0.481 0.269 0.269 ...
#>  $ coalition: num  1 0 0 0 0 0 NA 1 1 1 ...
#>  $ enph     : num  3.41 2.34 2.34 1.75 1.75 ...
#>  $ enps     : num  2.04 2.04 2.04 1.91 1.91 ...
#>  $ c_h      : num  0.796 0.931 0.931 0.608 0.608 ...
```

## Installation

To install the development version from GitHub:

``` r
if (!require("remotes")) install.packages("remotes")
remotes::install_github("Nicolas-Schmidt/partiesAL")
```

## Example

In the following example, the `parties` database is used to visualize
the temporal evolution of the number of political parties in the
congresses of each country.

``` r
library(partiesAL)

nparty <- 
    partiesAL::parties %>% 
    select(ccode, pty_code, year, legis) %>% 
    filter(legis %in% c(1,3)) %>% 
    distinct() %>%
    select(-pty_code, - legis) %>% 
    split(., .$ccode) %>% 
    lapply(., table) %>% 
    lapply(., as.data.frame) %>% 
    do.call('rbind', .) 


ggplot(nparty, aes(x = as.numeric(as.character(year)), y = Freq))+
    geom_bar(stat="identity", width = NULL, fill = "red", color = "black")+ 
    facet_wrap(~ccode, ncol = 3) + 
    theme_minimal() +
    labs(x = "", y = "")
```

<img src="man/figures/README-unnamed-chunk-6-1.png" width="100%" />


NEWS.md
# partiesAL 0.1.0

* Release version


R/parties2.R
#' parties2
#'
#'
#'\describe{
#'    \item{cowcode}{Country code (from Correlates of War Project).}
#'    \item{ccode}{Country code (International Standards Organization, Alpha-3 code)}
#'          \describe{
#'              \item{Argentina}{ARG}
#'              \item{Bolivia}{BOL}
#'              \item{Brazil}{BRA}
#'              \item{Chile}{CHL}
#'              \item{Colombia}{COL}
#'              \item{Costa Rica}{CRI}
#'              \item{Cuba}{CUB}
#'              \item{Dominican Republic}{DOM}
#'              \item{Ecuador}{ECU}
#'              \item{El Salvador}{SLV}
#'              \item{Guatemala}{GTM}
#'              \item{Haiti}{HTI}
#'              \item{Honduras}{HND}
#'              \item{Mexico}{MEX}
#'              \item{Nicaragua}{NIC}
#'              \item{Panama}{PAN}
#'              \item{Paraguay}{PRY}
#'              \item{Peru}{PER}
#'              \item{Uruguay}{URY}
#'              \item{Venezuela}{VEN}
#'              \item{United States}{USA}
#'          }
#'    \item{year}{Year}
#'    \item{legis}{Legislature in operation?}
#'          \describe{
#'              \item{0}{No (Congress was closed)}
#'              \item{1}{Yes (Congress in operation by the end of the year)}
#'              \item{2}{Closed worked for part of the year, but it was shut down before Dec. 31st}
#'              \item{3}{Constitutional Assembly in functions (but no regular congress)}
#'          }
#'    \item{date_low}{Date of the election for the Lower House}
#'    \item{date_upp}{Date of the election for the Upper House}
#'    \item{pty_acrn}{Party acronym}
#'    \item{pty_name}{Party name (full party name, as reported by Political Handbook of the World or by your other sources).}
#'    \item{pty_code}{Nine-digit party code: The first three digits match the country code; the next four digits are the year in which the party was founded; the last two digits identify the specific party, in case more than one party was created in the same country and the same year. For example, the Radical Party (a.k.a. Union Civica Radical or UCR) was founded in Argentina (cowcode = 160) in 1890. Thus, the code for the party is 160189001.}
#' }
#' @docType data
#' @keywords datasets
#' @name parties2
#' @usage data(parties2)
#' @format A data frame with 1995 rows and 17 variables
#' @references Aníbal Pérez-Liñan and Nicolás Schmidt and Daniela Vairo (2019), Presidential hegemony and democratic backsliding in Latin America, 1925-2016, Democratization, 26, 4, pp.606-625, https://doi.org/10.1080/13510347.2019.1566321.
#' @source empty
NULL


R/parties.R
#' parties
#'
#'
#'\describe{
#'    \item{cowcode}{Country code (from Correlates of War Project).}
#'    \item{ccode}{Country code (International Standards Organization, Alpha-3 code)}
#'          \describe{
#'              \item{Argentina}{ARG}
#'              \item{Bolivia}{BOL}
#'              \item{Brazil}{BRA}
#'              \item{Chile}{CHL}
#'              \item{Colombia}{COL}
#'              \item{Costa Rica}{CRI}
#'              \item{Cuba}{CUB}
#'              \item{Dominican Republic}{DOM}
#'              \item{Ecuador}{ECU}
#'              \item{El Salvador}{SLV}
#'              \item{Guatemala}{GTM}
#'              \item{Haiti}{HTI}
#'              \item{Honduras}{HND}
#'              \item{Mexico}{MEX}
#'              \item{Nicaragua}{NIC}
#'              \item{Panama}{PAN}
#'              \item{Paraguay}{PRY}
#'              \item{Peru}{PER}
#'              \item{Uruguay}{URY}
#'              \item{Venezuela}{VEN}
#'              \item{United States}{USA}
#'          }
#'    \item{year}{Year}
#'    \item{legis}{Legislature in operation?}
#'          \describe{
#'              \item{0}{No (Congress was closed)}
#'              \item{1}{Yes (Congress in operation by the end of the year)}
#'              \item{2}{Closed worked for part of the year, but it was shut down before Dec. 31st}
#'              \item{3}{Constitutional Assembly in functions (but no regular congress)}
#'          }
#'    \item{date_low}{Date of the election for the Lower House}
#'    \item{date_upp}{Date of the election for the Upper House}
#'    \item{pty_acrn}{Party acronym}
#'    \item{pty_name}{Party name (full party name, as reported by Political Handbook of the World or by your other sources).}
#'    \item{pty_code}{Nine-digit party code: The first three digits match the country code; the next four digits are the year in which the party was founded; the last two digits identify the specific party, in case more than one party was created in the same country and the same year. For example, the Radical Party (a.k.a. Uni?n C?vica Radical or UCR) was founded in Argentina (cowcode = 160) in 1890. Thus, the code for the party is 160189001.}
#'    \item{fac_name}{Name of the faction.  Factions may lack a formal name; in that case, is added a brief note to identify the faction (e.g., "Dissident Liberals"). If the row represents a whole party (which is the norm) the variable is left blank.}
#'    \item{fac_code}{Twelve-digit faction code: The first nine digits are equal to the party code (pty_code); the last three digits are an arbitrary sequence to identify the specific faction within the party (e.g., 001). }
#'    \item{s_low}{Number of seats obtained by the party (faction) in the Lower (only) House. If there is no legislature, or if the party has no seats in the lower House, the variable is left blank.}
#'    \item{ts_low}{Total number of seats in the Lower (or only) House (including vacant seats). If no legislature, it is left blank.}
#'    \item{s_upp}{Number of seats obtained by the party (faction) in the Upper House. Note: If there is no Senate, or if the party has no seats in the Senate, it is left blank.}
#'    \item{ts_upp}{Total number of seats in the Upper House (including vacant seats). Note: If no legislature, it is left blank.}
#'    \item{presp}{Coded 1 if the line corresponds to the president's party; 0 otherwise.}
#'    \item{cl_other}{Coded 1 if party belongs to the president's coalition; 0 otherwise. Coalitions are blocs that consistently support the president in the legislature. Different sources are used to define which parties (or factions) supported the president.}
#'    \item{cl_altman}{Coalitions coded as is in: Altman, David. 2001. The Politics of Coalition Formation and Survival in Multiparty Presidential Regimes. Ph.D. Dissertation, University of Notre Dame.}
#'    \item{cl_Deheza}{Coalitions coded as is in: Deheza, Grace Ivana. 1997. Gobiernos de coalici?n en el sistema presidencial: America del Sur, Departamento de Ciencias Pol?ticas y Sociales, Instituto Universitario Europeo, Florencia.}
#'    \item{cl_DPI}{Coalitions coded as is in: The Database of Political Institutions (DPI). Available in: https://publications.iadb.org/handle/11319/8806}
#'    \item{founded}{Year in which the party was founded. We use 8888 as a default.}
#'    \item{source}{Source for the information reported.}
#' }
#' @docType data
#' @keywords datasets
#' @name parties
#' @usage data(parties)
#' @format A data frame with 11822 rows and 22 variables
#' @references Aníbal Pérez-Liñan and Nicolás Schmidt and Daniela Vairo (2019), Presidential hegemony and democratic backsliding in Latin America, 1925-2016, Democratization, 26, 4, pp.606-625, https://doi.org/10.1080/13510347.2019.1566321.
#' @source empty
NULL
